solutions/community-deploy/communitybuild.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+usage() { 
+echo "Generate cortx-all docker image from provided CORTX release build"
+echo "Usage: $0 [ -b brunch ] [ -p push docker-image to GHCR yes/no. Default no] [ -t tag latest yes/no. Default no" ] [ -e environment ] [ -h print help message ] 1>&2; exit 1; }
+while getopts "b:h:" opt; do
+    case $opt in
+        b ) BRANCH=$OPTARG;;
+        h ) usage
+        exit 0;;
+        *) usage
+        exit 1;;
+    esac
+done
+
+function DockerCheck() {
+        docker --version >/dev/null 2>&1
+        if [ $? -eq 0 ]; then
+                echo "Docker is installed and $(docker --version)."
+        else
+                echo "Docker not installed"
+                printf "\n################################################################################\n"
+                printf "\tInstalling Docker Engine\n"
+                printf "\n################################################################################\n"
+                yum install -y yum-utils
+                yum-config-manager \
+    --add-repo \
+    https://download.docker.com/linux/centos/docker-ce.repo -y
+                yum install -y docker-ce docker-ce-cli containerd.io docker-compose-plugin
+                systemctl start docker
+                docker --version
+#               exit 1
+        fi
+}
+DockerCheck
+
+function DockerComposeCheck() {
+        docker-compose --version >/dev/null 2>&1
+        if [ $? -eq 0 ]; then
+                echo "Docker-Compose is installed and $(docker-compose --version)."
+        else
+                echo "Docker-Compose not installed"
+                printf "\n################################################################################\n"
+                printf "\tInstalling Docker-Compose\n"
+                printf "\n################################################################################\n"
+                curl -SL https://github.com/docker/compose/releases/download/v2.5.0/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose
+                chmod +x /usr/local/bin/docker-compose
+                ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
+                docker-compose --version
+#               exit 1
+        fi
+}
+DockerComposeCheck
+
+# Compile and Build CORTX Stack
+docker pull ghcr.io/seagate/cortx-build:rockylinux-8.4
+
+# Clone the CORTX repository
+cd /mnt && git clone https://github.com/Seagate/cortx --recursive --depth=1
+
+# Checkout main branch for generating CORTX packages
+docker run --rm -v /mnt/cortx:/cortx-workspace ghcr.io/seagate/cortx-build:rockylinux-8.4 make checkout BRANCH=$BRANCH
+
+# Validate CORTX component clone status
+cd /mnt/cortx/ 
```

Comment:
use `pushd` instead of cd to directory and then `popd` for getting back to original directory instead of using `cd -`